### PR TITLE
App config unit testing

### PR DESCRIPTION
**Pull request by Rkareko**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    private lateinit var fhirPathDataExtractor: FhirPathDataExtractor
+    private lateinit var configurationRegistry: ConfigurationRegistry
+    private lateinit var rulesFactory: RulesFactory
+    private lateinit var rulesEngineService: RulesEngineService
+
+    @Before
+    fun setUp() {
+        configurationRegistry = Faker.buildTestConfigurationRegistry(mockk())
+        fhirPathDataExtractor = mockk(relaxed = true)
+        rulesEngine = mockk()//(relaxed = true)
```
Comment (FikriMilano) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt]: is the comment still valid?

**Review by FikriMilano — APPROVED** [review]
Good stuff Richard

File: android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt (hunk 1 above)
Comment (Rkareko) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt]: No. Will remove

File: android/engine/src/test/java/org/smartregister/fhircore/engine/p2p/dao/BaseP2PTransferDaoTest.kt [hunk 2]
```diff
@@ -253,6 +253,33 @@ class BaseP2PTransferDaoTest : RobolectricTest() {
     assertEquals("Patient", search.type.name)
   }
 
+  @Test
+  fun `getDefaultDataTypes() returns correct list of datatypes`() {
```
Comment (ellykits) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/p2p/dao/BaseP2PTransferDaoTest.kt:257]: Can be renamed as per the recommendation in the [Code-Testing- wiki](https://github.com/opensrp/fhircore/wiki/Code-Testing)

File: android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      spyk(
+        RulesFactory(
+          configurationRegistry = configurationRegistry,
+          rulesEngineService = rulesEngineService
+        )
+      )
+  }
+
+  @Test
+  fun `init() populates facts`() {
```
Comment (ellykits) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt:75]: Can be renamed as per the recommendation in the [Code-Testing- wiki](https://github.com/opensrp/fhircore/wiki/Code-Testing)

File: android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    var facts = ReflectionHelpers.getField<Facts>(rulesFactory, "facts")
+    Assert.assertEquals(2, facts.asMap().size)
+    Assert.assertNotNull(facts.get("data"))
+    Assert.assertNotNull(facts.get("fhirPath"))
+  }
+
+  @Test
+  fun `beforeEvaluate() returns true`() {
```
Comment (ellykits) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt:83]: Can be renamed as per the recommendation in the [Code-Testing- wiki](https://github.com/opensrp/fhircore/wiki/Code-Testing)

File: android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    Assert.assertTrue(rulesFactory.beforeEvaluate(mockk(), mockk()))
+  }
+
+  @Test
+  fun `fireRule() calls rulesEngine#fire`() {
```
Comment (ellykits) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/rulesengine/RulesFactoryTest.kt:88]: Can be renamed as per the recommendation in the [Code-Testing- wiki](https://github.com/opensrp/fhircore/wiki/Code-Testing)

**Review by ellykits — CHANGES_REQUESTED** [review]

**Review by ellykits — APPROVED** [review]

**Pull request merged**